### PR TITLE
Allow general search and replace in Preprocess

### PR DESCRIPTION
**Pull request by 2m**
Introduces `preprocessRules` which allows general search and replace by regex in `Preprocess`

Makes it possible to fix https://github.com/akka/alpakka/issues/1378

### CODE REVIEW

File: src/main/scala/com/typesafe/sbt/site/preprocess/PreprocessKeys.scala [hunk 1]
```diff
@@ -10,5 +14,6 @@ trait PreprocessKeys {
       "preprocessIncludeFilter", "Filter defining set of files to preprocess")
   val preprocessVars = SettingKey[Map[String, String]](
       "preprocessVars", "Replacements for preprocessing.")
+  val preprocessRules = SettingKey[Seq[(Regex, Match => String)]]("preprocessRules", "Rules for proprocessing")
```
Comment (ennru) [inline src/main/scala/com/typesafe/sbt/site/preprocess/PreprocessKeys.scala]:
```suggestion
   val preprocessRules = SettingKey[Seq[(Regex, Match => String)]]("preprocessRules", "Rules for preprocessing supporting Regular Expressions")
```

**Comment by ennru** [comment]
Would be good to exercise this feature in scripted tests.

**Comment by 2m** [comment]
Added scripted test and documentation.

Needed to upgrade paradox theme to be able to generate project documentation locally.

**Comment by 2m** [comment]
Pushed a scripted test with a useful preprocessing rule.

**Comment by 2m** [comment]
This does not fail locally for me. Investigating...

**Comment by 2m** [comment]
Locally it worked, because I had files in the `target` directory, and those were copied over by the `scripted` task.

**Pull request merged**